app/controllers/unaccompanied_controller.rb

```diff
@@ -504,6 +505,24 @@ def remove_adult
     end
   end
 
+  def remove_other_sponsor_name
```

Comment:
@samuele-mattiuzzo - we should discuss moving this behaviour out of the controller into the UAM model to reduce the controller responsibilities. 
I haven't done this yet because 1) It works and we can get it out there, 2) It's a different strategy to the rest of the code.